bin/007

```diff
@@ -33,6 +48,16 @@ sub run_007($program, Str $backend is copy) {
     exit($runtime.exit-code);
 }
 
+multi MAIN(Str :$backend = "default") {
+    my $rlwrap-installed = "" ne qx[which rlwrap].trim;
+    if !$rlwrap-installed {
+        note "rlwrap not found in PATH. (Do you have it installed?)";
+        exit(1);
+    }
+
+    shell "rlwrap perl6 -e '{$REPL_PROGRAM}'";
```

Comment:
are we really doing it this way? :-)

Reply:
Yes. :smile: As usual, I might be missing a much better solution (that still gives us history and all the navigation keyboard shortcuts), in which case, please let me know! Preferably through a PR.

For what it's worth, I'm well aware of the limitations of this approach. Each line is its own compunit, for example, so doing `my name = "fred"` on one line and then `say(name)` on the next won't work.

Still, I think as a compromise, this is better than _no_ REPL, and that's why I eagerly added this binary (copied from someone else, I believe) back in 2016. Before I saw this solution, I didn't believe even such a compromise existed. See charter declaration in 9c7c4f491bdf5c6d438dec8fd22cace18981d917.